app/controllers/custom/polls_controller.rb

```diff
@@ -2,20 +2,69 @@
 
 class PollsController < ApplicationController
   include GuestUsers
+  before_action :set_answers, only: [:answer, :show]
+  before_action :set_questions, only: [:answer, :show]
+  before_action :set_question_answers, only: [:answer, :show]
+  before_action :set_token, only: [:answer, :show]
+  before_action :set_commentable, only: [:answer, :show]
+
+  invisible_captcha only: [:answer], honeypot: :title
 
   def show
-    @questions = @poll.questions.for_render.sort_for_list
-    @token = poll_voter_token(@poll, current_user)
-    @poll_questions_answers = Poll::Question::Answer.where(question: @poll.questions)
-                                                    .with_content.order(:given_order)
+  end
 
-    @answers_by_question_id = {}
-    poll_answers = ::Poll::Answer.by_question(@poll.question_ids).by_author(current_user&.id)
-    poll_answers.each do |answer|
-      @answers_by_question_id[answer.question_id] = answer.answer_id
+  def answer
+    if @poll_answers.map(&:valid?).all?(true)
```

Comment:
Hi @javierm, finally kept this line as it was because I want to call the `valid?` method on all the answers, otherwise the `all?(&:valid?)` stops when finding the first invalid answer skipping validation of the following answers.